scaleway/resource_iam_group_test.go

```diff
@@ -0,0 +1,485 @@
+						}
+						resource "scaleway_iam_application" "app02" {
+							name = "second app"
+						}
+						resource "scaleway_iam_group" "main_app" {
+							name = "iam_group_app"
+							application_ids = [
+								scaleway_iam_application.app01.id,
+								scaleway_iam_application.app02.id,
+							]
+						}
+					`,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckScalewayIamGroupExists(tt, "scaleway_iam_group.main_app"),
+					resource.TestCheckResourceAttr("scaleway_iam_group.main_app", "name", "iam_group_app"),
+					resource.TestCheckResourceAttr("scaleway_iam_group.main_app", "application_ids.#", "2"),
+					resource.TestCheckResourceAttrPair("scaleway_iam_group.main_app", "application_ids.0", "scaleway_iam_application.app01", "id"),
+					resource.TestCheckResourceAttrPair("scaleway_iam_group.main_app", "application_ids.1", "scaleway_iam_application.app02", "id"),
+				),
+			},
+			{
+				Config: `
+						resource "scaleway_iam_application" "app01" {
+							name = "first app"
+						}
+						resource "scaleway_iam_application" "app02" {
+							name = "second app"
+						}
+						resource "scaleway_iam_group" "main_app" {
+							name = "iam_group_app"
+							application_ids = [
+								scaleway_iam_application.app02.id,
+							]
+						}
+					`,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckScalewayIamGroupExists(tt, "scaleway_iam_group.main_app"),
+					resource.TestCheckResourceAttr("scaleway_iam_group.main_app", "name", "iam_group_app"),
+					resource.TestCheckResourceAttr("scaleway_iam_group.main_app", "application_ids.#", "1"),
+					resource.TestCheckResourceAttrPair("scaleway_iam_group.main_app", "application_ids.0", "scaleway_iam_application.app02", "id"),
+				),
+			},
+			{
+				Config: `
+						resource "scaleway_iam_application" "app01" {
+							name = "first app"
+						}
+						resource "scaleway_iam_application" "app02" {
+							name = "second app"
+						}
+						resource "scaleway_iam_group" "main_app" {
+							name = "iam_group_app"
+						}
+					`,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckScalewayIamGroupExists(tt, "scaleway_iam_group.main_app"),
+					resource.TestCheckResourceAttr("scaleway_iam_group.main_app", "name", "iam_group_app"),
+					resource.TestCheckResourceAttr("scaleway_iam_group.main_app", "application_ids.#", "0"),
+					resource.TestCheckNoResourceAttr("scaleway_iam_group.main_app", "application_ids.0"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccScalewayIamGroup_Users(t *testing.T) {
+	SkipBetaTest(t)
+	tt := NewTestTools(t)
+	defer tt.Cleanup()
+	resource.ParallelTest(t, resource.TestCase{
+		ProviderFactories: tt.ProviderFactories,
+		CheckDestroy: resource.ComposeTestCheckFunc(
+			testAccCheckScalewayIamGroupDestroy(tt),
+		),
+		Steps: []resource.TestStep{
+			{
+				Config: `
+						resource "scaleway_iam_group" "main_user" {
+							name = "iam_group_user"
+							user_ids = [
+								"29c31dd4-8ea1-4927-82d9-a0620e04773f"
```

Comment:
Can we have a data source about this? It would be nice to use something human-readable to understand who the user is.